viewing_party/party.py

```diff
@@ -1,23 +1,226 @@
 # ------------- WAVE 1 --------------------
+import json
+from unittest import result
 
 def create_movie(title, genre, rating):
-    pass
+    '''
+    Input: movie title, genre, and rating
+    Ouput: If values truthy, produces a dictionary of movies with title, genre, and rating key-value pairs.
+    If inputs are falsy, function returns None.
+    '''
+    movie_dict = {}
+    # if title == True and genre == True and rating == True:
+    if True: 
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    if title == None or genre == None or rating == None:
+        return None
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    '''
+    Input: user_data is a dictionary with a key "watched" and value of list of movie dictionaries the user has watched
+    Output: Function adds movie dictionary to user_data and returns user_data
+    '''
+    user_data["watched"] = [movie]
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    Input: user_data is list of movie dictionaries with key "watchlist" and value of list of movie dictionaries
+    that the user wants to watch
+    Output: Function adds movie dictionary to "watchlist" and returns user_data
+    '''
+    user_data["watchlist"] = [movie]
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    Input: user dictionary is a list of movie dictionaries with "watched" and "watchlist keys
+    representing the movies the user has watched or wants to watch. Title is a string representing
+    a movie the user has watched. 
+    Output: function moves movie represented by title to watched list if the user has that movie in the watchlist.
+    Function returns user_data.
+    '''
+    for movie_index in range(len(user_data["watchlist"])):
+        for title_key,current_title in user_data["watchlist"][movie_index].items():
+            if current_title == title:

```

Comment:
The power of dictionaries is that we can use their keys to find values in O(1) time. So rather than looping over each dictionary item to find the title, we can just use the title key.

With the previous suggestion to iterate directly over the movie value, we could refactor this function to be:
```
for movie in user_data["watchlist"]:
    if movie["title"] == title:
        user_data["watched"].append(movie)
        user_data["watchlist"].remove(movie)
```
(Using `remove` instead of `pop` to allow for deleting by value instead of index)